src/data/int/cast.lean

```diff
@@ -270,6 +270,10 @@ lemma int_apply (n : ℤ) (a : α) : (n : ∀ a, β a) a = n := rfl
 
 end pi
 
+lemma sum.elim_int_cast_int_cast {α β γ : Type*} [has_int_cast γ] (n : ℤ) :
+  sum.elim (n : α → γ) (n : β → γ) = n :=
+by { ext x, cases x; simp }
```

Comment:
I suspect every `simp` in this PR can be replaced with `refl`

Reply:
Could you please add a more general `sum.elim_const_const : sum.elim (const _ c) (const _ c) = const _ c`? Probably, both a version with `function.const` and a version with lambda functions. This lemma, `elim_nat_cast_nat_cast`, and `elim_one_one` are special cases. BTW, these four lemmas (incl. `elim_zero_zero`) should be `@[simp]`.

Reply:
Now that you mention other function constructors, we should probably have `sum.elim (single i x) 0 = single (sum.inl i) x` and the reverse; and similarly for `update`